universal-login-react/src/App.tsx

```diff
@@ -59,7 +60,7 @@ export const App = () => {
     return () => {};
   };
 
-  const [deployedWallet, setDeployedWallet] = useState(new DeployedWallet('', '', '', sdk));
+  const [walletService] = useState(new WalletService(sdk));
```

Comment:
Why not just `const walletService = new WalletService(sdk)` ?

Reply:
👍

Reply:
> Why not just const walletService = new WalletService(sdk) ?

That way, for each re-render new `WalletService` would be created

Reply:
Are you sure that with `useState`, new `WalletService` will be created more rarely?

I am new to functional components and hooks so I don't know, but quick experiment with `console.log` in `WalletService`'s constructor shows that new instance is created with `useState` as well.

Reply:
hmm, thats odd, could you send me an example?

Reply:
On a second thought: the code above will actually create a new wallet service every render, but that wallet service is **discarded** on subsequent renders

Reply:
I've approved and it is not important, but I think the conversation is interesting. What do you mean "discarded"? Are you talking about garbage collection?

Reply:
@rzadp 
Lets compare the two pieces of code (assume its inside a render method):
```
// 1
const walletService = new WalletService()

// 2
const [walletService] = useState(new WalletService())
```

Although in both example a new instance of `WalletService` is created each render, the variable  `walletService`  will always hold the same instance in the second example, while containing a new value every render in a first example.

There is also a special syntax to avoid needlessly creating a new instance of `WalletService` every render:

```
const [walletService] = useState(() => new WalletService())
```

React will call the lambda only once: on the initial render.

Reply:
OK I think I get it, thanks.